nibabel/gifti/parse_gifti_fast.py

```diff
@@ -309,64 +321,27 @@ def flush_chardata(self):
             # update the endianness according to the
             # current machine setting
             self.endian = gifti_endian_codes.code[sys.byteorder]
+
         elif self.write_to == 'Label':
             self.label.label = data.strip()
 
     @property
     def pending_data(self):
         " True if there is character data pending for processing "
-        return not self._char_blocks is None
+        return self._char_blocks is not None
 
 
-def parse_gifti_file(fname=None, fptr=None, buffer_size=None):
-    """ Parse gifti file named `fname`, return image
-
-    Parameters
-    ----------
-    fname : str
-        filename of gifti file
-    buffer_size: None or int, optional
-        size of read buffer. None gives default of 35000000 unless on python <
-        2.6, in which case it is read only in the parser.  In that case values
-        other than None cause a ValueError on execution
-
-    Returns
-    -------
-    img : gifti image
-    """
-    assert (fname is not None) + (fptr is not None) == 1, "Specify only fname or fptr, not both"
-
-    if fptr is None:
-        with open(fname, 'rb') as datasource:
-            return parse_gifti_file(fptr=datasource, buffer_size=buffer_size)
-    else:
-        datasource = fptr
+class Outputter(GiftiImageParser):
+    @np.deprecate_with_doc("Use GiftiImageParser instead.")
```

Comment:
I probably don't understand what deprecating the class would do; could you help make that clear to me?

In previous cases I remember deprecating `__init__` so that the deprecation is lazy--it only triggers if someone instantiates the class, and not when the file is simply imported. That's the only goal here.